File: android/UClass/app/src/main/java/com/ubase/uclass/network/response/BaseData.kt
package com.ubase.uclass.network.response

import com.google.gson.annotations.SerializedName

/**
 * 기본 API 응답 데이터 모델 (제네릭 버전)
 */
data class BaseData<T>(
    @SerializedName("success")
    val isSuccess: Boolean,

    @SerializedName("message")
    val message: String? = null,

    @SerializedName("data")
    val data: T? = null,

    @SerializedName("timestamp")
    val timestamp: String? = null
)

/**
 * 빈 데이터용 클래스
 */
class EmptyData

File: ios/uclass/uclass/network/response/BaseData.swift
/**
 * 기본 API 응답 데이터 모델 (제네릭 버전)
 */
struct BaseData<T: Codable>: Codable {
    let isSuccess: Bool
    let message: String?
    let data: T?
    let timestamp: String?
    
    private enum CodingKeys: String, CodingKey {
        case isSuccess = "success"
        case message
        case data
        case timestamp
    }
}


struct EmptyData: Codable {}
